torch/export/dynamic_shapes.py

```diff
@@ -1064,10 +964,10 @@ def _get_dim_name_mapping(
             continue
         if isinstance(dim, int):
             continue
-        assert isinstance(dim, _Dim)  # dim hints should have boiled away
-        name_to_dim[dim.__name__] = dim
-        if isinstance(dim, _DerivedDim):
-            name_to_dim[dim.root.__name__] = dim.root  # type: ignore[attr-defined]
+        if isinstance(dim, _Dim):
```

Comment:
`else` what? Maybe handle that explicitly with a pass and then assert / give a useful error if anything unexpected happens?